server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -552,6 +586,17 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         }
     }
 
+    private DataStream updateDataStream(DataStream dataStream, Metadata.Builder metadata, RestoreSnapshotRequest request) {
```

Comment:
Maybe write a unit test for this method, which tests the renaming more thoroughly?

Reply:
Done